tests/manage/mcg/test_virtual_hosted_buckets.py

```diff
@@ -0,0 +1,67 @@
+import logging
+
+import pytest
+
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.bucket_utils import verify_s3_object_integrity
+from ocs_ci.framework.pytest_customization.marks import bugzilla, tier2, polarion_id
+
+logger = logging.getLogger(__name__)
+
+
+@tier2
+@bugzilla("2183092")
+@polarion_id("OCS-5217")
+class TestVirtualHostedBuckets:
+    @pytest.fixture()
+    def s3cmd_setup(self, awscli_pod_session):
+        """
+        Setup s3cmd tool in s3cli pod
+
+        """
+        awscli_pod_session.exec_cmd_on_pod(command="apk add s3cmd")
+        return awscli_pod_session
+
+    def test_virtual_hosted_bucket(self, s3cmd_setup, bucket_factory):
+        """
+        Test s3 operations on virtual style hosted buckets
+
+        """
+        logger.info("Creating obc")
+        bucket_name = bucket_factory()[0].name
+
+        # create a route for the bucket create above
+        s3_route_data = OCP(
+            kind="route",
+            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+            resource_name="s3",
+        ).get()
+        s3_route_data["metadata"]["name"] = f"s3-{bucket_name}"
+        s3_route_data["spec"]["host"] = f'{bucket_name}.{s3_route_data["spec"]["host"]}'
+        new_route_obj = OCS(**s3_route_data)
+        logger.info(f"Creating new route for bucket {bucket_name}")
+        new_route_obj.create()
+
+        host_base = f'{s3_route_data["spec"]["host"]}'
+        host_bucket = f'%(bucket)s.{s3_route_data["spec"]["host"]}'
+
+        orginal_path = "/test_objects/random1.txt"
+        downloaded_path = "/aws/random1.txt"
```

Comment:
To avoid leaving the downloaded file as a leftover in the pod's home directory, please use the test_directory_setup fixture. 

Also consider using `write_random_objects_in_pod` to generate a random object instead of relying on `random1.txt`.

Reply:
ack